### ISSUE TITLE
Changing getJavaClassName() for Mode.ENUM in DefaultGeneratorStrategy generates invalid code

### ISSUE THREAD

**Issue by chrisawad** [jOOQ/jOOQ#9903]
### Expected behavior and actual behavior:
Specifying a Java Class Name override in a generator strategy should produce methods that take in the newly named classes, but renaming Mode.ENUM class names currently breaks generated code.
 
### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

1. Create a class the extends DefaultGeneratorStrategy
2. Override getJavaClassName() st it remove all "_" from class names for EnumTypes:
    ```java
    @Override
    public String getJavaClassName(Definition definition, Mode mode) {
        return mode == Mode.ENUM 
                ? definition.getOutputName().replaceAll("_", "") 
                : super.getJavaClassName(definition, mode);
    }
    ``` 
3. Run the generator with your GeneratorStrategy
4. Notice EnumType classes are generated without the "_", which is what we want
5. Notice compile errors in other classes with methods that take in EnumType parameters. Those parameters still have "_".
### Versions:

- jOOQ: 3.12.1
- Java: 11
- Database (include vendor): MySQL 8.0
- OS: Fedora 30
- JDBC Driver (include name if inofficial driver): com.mysql.jdbc.Driver

**Comment by lukaseder** [comment]
Thanks for your report. I can reproduce this. Will fix immediately.

**Comment by lukaseder** [comment]
The problem seems to be that the `Mode.ENUM` flag is not passed consistently, which should be fixed, of course.

In the meantime, your implementation might be more robust if you wrote it like this:

```java
@Override
public String getJavaClassName(Definition definition, Mode mode) {
    return definition instanceof EnumDefinition // Fix here
            ? definition.getOutputName().replaceAll("_", "") 
            : super.getJavaClassName(definition, mode);
}
```

**Comment by lukaseder** [comment]
Fixed for 3.14.0. Backport scheduled for 3.13.2 (#9909) and 3.12.5 (#9910)
